src/server/game/Entities/Item/ItemTemplate.cpp

```diff
@@ -139,6 +139,32 @@ uint32 ItemTemplate::GetSkill() const
     }
 }
 
+bool ItemTemplate::IsWeaponOfSubClass(ItemSubclassWeapon subClass) const
+{
+    if (Class != ITEM_CLASS_WEAPON)
+        return false;
+
+    return SubClass == subClass; // here we got a sign unsigned error iirc
```

Comment:
not sure if smth like this will work here: return SubClass == static_cast<uint32>(subClass);